Makefile.am

```diff
@@ -43,32 +43,37 @@ MAINTAINERCLEANFILES = autom4te.cache
 
 # for distcheck
 clean-local:
-	rm -f packcc$(BUILD_EXEEXT)
 	@if test "$(top_srcdir)" != "$(top_builddir)"; then \
 		rm -rf $(OPTLIB2C_SRCS); \
 		rm -rf $(TXT2CSTR_SRCS); \
 	else \
 		$(MAKE) -C docs clean; \
 	fi
 
+maintainer-clean-local:
+	rm -rf .deps misc/packcc/.dirstamp
+
 bin_PROGRAMS = ctags
 noinst_LIBRARIES = libctags.a
 
 noinst_PROGRAMS =
 noinst_PROGRAMS += mini-geany
 
+# packcc always uses native compiler even when cross-compiling.
+# packcc cannot use the standard Automake rule.
 PACKCC_CPPFLAGS_FOR_BUILD  = -fsigned-char
 if HAVE_STRNLEN_FOR_BUILD
 PACKCC_CPPFLAGS_FOR_BUILD += -DPCC_USE_SYSTEM_STRNLEN
 endif
 
 PACKCC = $(top_builddir)/packcc$(BUILD_EXEEXT)
+CLEANFILES += $(PACKCC)
 
 cc4b_verbose = $(cc4b_verbose_@AM_V@)
 cc4b_verbose_ = $(cc4b_verbose_@AM_DEFAULT_V@)
 cc4b_verbose_0 = @echo CC4BUILD "  $@";
-$(PACKCC): $(top_srcdir)/misc/packcc/src/packcc.c
-	$(cc4b_verbose)$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) $(PACKCC_CPPFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $(top_srcdir)/misc/packcc/src/packcc.c
+$(PACKCC): $(top_srcdir)/$(PACKCC_SRCS)
```

Comment:
I strongly expect the number of the source file(s) of packcc is one even in the future.
However, the name `PACKCC_SRCS` implies it can hold more than one item though it holds only one item actually.

So, how about doing like:
```Make
$(PACKCC): $(foreach let,$(PACKCC_SRCS),$(top_srcdir)/$(let))
```
Ideally, we should not use GNU Make extension.

Reply:
> However, the name PACKCC_SRCS implies it can hold more than one item though it holds only one item actually.

I made this change to support the case where PACKCC_SRCS hold more than one item.  But you are right, I overlooked the concatenation of `$(top_srddir)/`.

> Ideally, we should not use GNU Make extension. but I cannot find a way without the extension.

If we can use GNU Make extension, we can simply fix as follows.  (Use VPATH and `$<`)

```diff
- $(PACKCC): $(top_srcdir)/misc/packcc/src/packcc.c
- 	$(cc4b_verbose)$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) $(PACKCC_CPPFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $(top_srcdir)/misc/packcc/src/packcc.c
+ $(PACKCC): $(PACKCC_SRCS)
+ 	$(cc4b_verbose)$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) $(PACKCC_CPPFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $<
```

Actually I used `$<' when I pushed first. But it failed on some platforms, because use of `$<` does not work on some platforms. (See [12.1 $< in Ordinary Make Rules](https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.70/autoconf.html#g_t_0024_003c-in-Ordinary-Make-Rules)),

How about the following fix? (Add `$(top_srcdir)/` at the definition of `$(PACKCC_SRCS)`)

```diff
diff --git a/Makefile.am b/Makefile.am
index 22b3a4bb..5cffa241 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -71,8 +71,8 @@ CLEANFILES += $(PACKCC)
 cc4b_verbose = $(cc4b_verbose_@AM_V@)
 cc4b_verbose_ = $(cc4b_verbose_@AM_DEFAULT_V@)
 cc4b_verbose_0 = @echo CC4BUILD "  $@";
-$(PACKCC): $(top_srcdir)/$(PACKCC_SRCS)
-   $(cc4b_verbose)$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) $(PACKCC_CPPFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $(top_srcdir)/$(PACKCC_SRCS)
+$(PACKCC): $(PACKCC_SRCS)
+ $(cc4b_verbose)$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CPPFLAGS_FOR_BUILD) $(PACKCC_CPPFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $(PACKCC_SRCS)

 if USE_READCMD
 bin_PROGRAMS+= readtags
diff --git a/source.mak b/source.mak
index b4b1e677..48977d85 100644
--- a/source.mak
+++ b/source.mak
@@ -420,8 +420,9 @@ READTAGS_HEADS = \
        $(NULL)
 READTAGS_OBJS  = $(READTAGS_SRCS:.c=.$(OBJEXT))

+# "$(top_srcdir)/" is required since VPATH does not work on this.
 PACKCC_SRCS = \
-   misc/packcc/src/packcc.c \
+ $(top_srcdir)/misc/packcc/src/packcc.c \
        \
        $(NULL)
 PACKCC_OBJS = $(PACKCC_SRCS:.c=.$(OBJEXT))
```

Reply:
I wonder whether mk_mingw.mak works well or not with this change.

Reply:
> I wonder whether mk_mingw.mak works well or not with this change.

With and without this change, both `mk_mingw.mak` and `mk_mvc.mak`  do not work on the case where PACKCC_SRCS holds more than one item. They assume a single file input.

mk_mingw.mak

```make
$(PACKCC_OBJS): $(PACKCC_SRCS)
	$(V_CC) $(CC_FOR_PACKCC) -c $(OPT) $(CFLAGS) $(COMMON_DEFINES) -o $@ $<

$(PACKCC): $(PACKCC_OBJS)
	$(V_CC) $(CC_FOR_PACKCC) $(OPT) -o $@ $^
```

The following should work.

```make
$(PACKCC): $(PACKCC_SRCS)
	$(V_CC) $(CC_FOR_PACKCC) $(OPT) $(CFLAGS) $(COMMON_DEFINES) -o $@ $^
```

`mk_mvc.mak` has the same issue.

BTW how about renaming `PACKCC_SRCS` to `PACK_SRC` and commenting that it must be a single file?

```diff
--- a/source.mak
+++ b/source.mak
@@ -420,12 +420,9 @@ READTAGS_HEADS = \
        $(NULL)
 READTAGS_OBJS  = $(READTAGS_SRCS:.c=.$(OBJEXT))

-PACKCC_SRCS = \
-   misc/packcc/src/packcc.c \
-   \
-   $(NULL)
-PACKCC_OBJS = $(PACKCC_SRCS:.c=.$(OBJEXT))
+# only one file can be specified.
+PACKCC_SRC = misc/packcc/src/packcc.c
+PACKCC_OBJ = $(PACKCC_SRC:.c=.$(OBJEXT))

 WIN32_HEADS = main/e_msoft.h
 WIN32_SRCS = win32/mkstemp/mkstemp.c
```

P.S.
I found I understood `$^` (all the prerequisite) and `$<` (the first prerequisite) oppositely.

Reply:
> BTW how about renaming PACKCC_SRCS to PACK_SRC and commenting that it must be a single file?

It is o.k. No comment is needed. The name tells everything developers should know.
Could you make a separate commit for renaming the macro names?